### PR TITLE
merge 2019.06.x

### PR DESCRIPTION
**Pull request by kkrebs**

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 811c7a5ad66ced05934a5916b9758b3771c12796! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5153-pull_request-891

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896&bid=15475843)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 8
           

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896&bid=15475843)

File: mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts [hunk 1]
```diff
@@ -48,7 +48,12 @@ namespace mycore.viewer {
 
             settings.filePath = encodeURI(settings.filePath);
 
-            if(settings.webApplicationBaseURL.lastIndexOf("/")==settings.webApplicationBaseURL.length-1){
+            if (settings.filePath.indexOf('_derivate_') > 0) {
+                settings.filePath = settings.filePath.replace('/', '%2F');
+            }
+
+            if(settings.webApplicationBaseURL !== "undefined"
+                && settings.webApplicationBaseURL.lastIndexOf("/") === settings.webApplicationBaseURL.length - 1){
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts:56]: Codacy Issue found: [" should be '](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts [hunk 2]
```diff
@@ -48,7 +48,12 @@ namespace mycore.viewer {
 
             settings.filePath = encodeURI(settings.filePath);
 
-            if(settings.webApplicationBaseURL.lastIndexOf("/")==settings.webApplicationBaseURL.length-1){
+            if (settings.filePath.indexOf('_derivate_') > 0) {
+                settings.filePath = settings.filePath.replace('/', '%2F');
+            }
+
+            if(settings.webApplicationBaseURL !== "undefined"
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts:55]: Codacy Issue found: [" should be '](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts:55]: Codacy Issue found: [missing whitespace](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts [hunk 3]
```diff
@@ -78,28 +80,45 @@ namespace mycore.viewer.widgets.iiif {
...
             //     return;
             // }
             //TODO Chaptertype currently not in Manifest
-            const chapterObject = new model.StructureChapter(parent,
-                '', this.getIDFromURL(chapter.id), chapter.getDefaultLabel());
-            // let chapterChildren = chapter.getRanges();
-
-            this.vChapterIdMap.set(chapterObject.id, chapterObject);
+            let chapterObject;
+            if (chapter === undefined || (chapter.getCanvasIds().length === 0 && chapter.getRanges().length === 0)) {
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts:89]: Codacy Issue found: [Unexpected use of undefined. (no-undefined)](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts [hunk 4]
```diff
@@ -78,28 +80,45 @@ namespace mycore.viewer.widgets.iiif {
...
+                chapterObject = new model.StructureChapter(parent,
+                    '', 'LOG_0', this.manifestDocument.getDefaultLabel());
+                this.vChapterIdMap.set(chapterObject.id, chapterObject);
+                cans.forEach((can: ICanvas) => {
+                    const childChap = new model.StructureChapter(chapterObject,
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts:94]: Codacy Issue found: ['model' is not defined. (no-undef)](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/iiif/widgets/IviewIIIFProvider.ts [hunk 5]
```diff
@@ -37,7 +38,7 @@ namespace mycore.viewer.widgets.iiif {
                 url: manifestDocumentLocation,
                 success: (response: any) => {
                     const manifest = <Manifest> manifesto.create(response);
-                    const builder = new IIIFStructureBuilder(<Manifest> manifest, tilePathBuilder);
+                    const builder = new IIIFStructureBuilder(<Manifest> manifest, tilePathBuilder, imageAPIURL);
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/iiif/widgets/IviewIIIFProvider.ts:41]: Codacy Issue found: ['IIIFStructureBuilder' is not defined. (no-undef)](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts [hunk 6]
```diff
@@ -78,28 +80,45 @@ namespace mycore.viewer.widgets.iiif {
...
+                        chapterObject, this.getIDFromURL(can.id), can.getDefaultLabel());
+                    chapterObject.chapter.push(childChap);
+                    this.vChapterIdMap.set(childChap.id, childChap);
+                });
+            }
+            else {
+                chapterObject = new model.StructureChapter(parent,
```
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts:101]: Codacy Issue found: ['model' is not defined. (no-undef)](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

File: mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts (hunk 1 above)
Comment (adlerre) [inline mycore-viewer/src/main/typescript/modules/base/MyCoReViewerSettings.ts:56]: Codacy Issue found: [missing whitespace](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)

**Pull request merged**